### PR TITLE
Return useful data from /pluginManager/api/json by default

### PR DESCRIPTION
**Pull request by himanshu748**
- Increase `@Exported` visibility to `2` for key `PluginWrapper` properties so they appear at default depth under `PluginManager.plugins`.
- Add `@Exported` to `PluginWrapper#getDisplayName()` as the non-deprecated alternative to `getLongName()`.
- Add an integration test validating that `/pluginManager/api/json` returns non-empty plugin objects and expected fields.

### CODE REVIEW

File: test/src/test/java/hudson/PluginManagerTest.java [hunk 1]
```diff
@@ -542,6 +542,37 @@ void pluginListJSONApi() throws Throwable {
         });
     }
 
+    @Issue("JENKINS-27993")
```
Comment (MarkEWaite) [inline test/src/test/java/hudson/PluginManagerTest.java]: The issue reference should point to the current issue tracker rather than the former issue tracker.
```suggestion
    @Issue("https://github.com/jenkinsci/jenkins/issues/21047")
```

File: core/src/main/java/hudson/PluginWrapper.java [hunk 2]
```diff
@@ -662,7 +663,7 @@ public String toString() {
      *
      * @deprecated For most purposes, use {@link #getDisplayName()}.
      */
-    @Exported
+    @Exported(visibility = 2)
```
Comment (MarkEWaite) [inline core/src/main/java/hudson/PluginWrapper.java]: I don't think we should show a deprecated attribute by default.  Why not leave this attribute only available if the caller requests additional depth?
```suggestion
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Thanks for the pull request and thanks for including a test of the changes.

There are two changes that I believe are needed.

File: core/src/main/java/hudson/PluginWrapper.java (hunk 2 above)
Comment (himanshu748) [reply core/src/main/java/hudson/PluginWrapper.java]: Feedback addressed: Removed @Exported(visibility = 2) from the deprecated getBackupVersion() and getLongName() methods in PluginWrapper.java.

**Review by daniel-beck — CHANGES_REQUESTED** [review]
Do not change lines unrelated to your pull request.

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Please don't force push changes.  It breaks the link between pull request comments and the commits.  We will squash merge at the end, so flawed steps along the way are not an issue.

Please remove the `@Exported(visibility = 2)` change from `getLongName` or explain why you must retain it.

**Review by himanshu748 — COMMENTED** [review]
Thanks for the review. I have removed `@Exported(visibility = 2)` from the deprecated `getLongName()` method, fixed the issue link in the test to point to GitHub, and updated the PR description to accurately reflect that `url` is not being asserted in this test since it requires update site metadata. Pushed without a force push.

File: test/src/test/java/hudson/PluginManagerTest.java [hunk 3]
```diff
@@ -263,9 +271,9 @@ public void startPlugin(PluginWrapper plugin) throws Exception {
         }
     }
 
-
     /**
-     * Makes sure that thread context classloader isn't used by {@link UberClassLoader}, or else
+     * Makes sure that thread context classloader isn't used by
```
Comment (daniel-beck) [inline test/src/test/java/hudson/PluginManagerTest.java]: No reformatting please. This applies to most changed lines in this file.

**Review by daniel-beck — DISMISSED** [review]

File: test/src/test/java/hudson/PluginManagerTest.java (hunk 3 above)
Comment (himanshu748) [reply test/src/test/java/hudson/PluginManagerTest.java]: Apologies for the noise. All unrelated formatting changes have been reverted in ca4cdf2532. The test file diff is now purely additive (+30/-0) — only the new test method, zero reformatting.

**Review by MarkEWaite — APPROVED** [review]
Thanks.  Change looks good to me

File: test/src/test/java/hudson/PluginManagerTest.java (hunk 1 above)
Comment (MarkEWaite) [reply test/src/test/java/hudson/PluginManagerTest.java]: Resolved in 919350144829e964b0a7309f377d165982f8c1fa

File: core/src/main/java/hudson/PluginWrapper.java (hunk 2 above)
Comment (MarkEWaite) [reply core/src/main/java/hudson/PluginWrapper.java]: Resolved in 919350144829e964b0a7309f377d165982f8c1fa
